### PR TITLE
Replace envoy_reloadable_features_successful_active_health_check_uneject_host with config

### LINKED ISSUE #29099: envoy_reloadable_features_successful_active_health_check_uneject_host deprecation
Your change #24331 (prefer active health check during outlier ejection wait) introduced a runtime guarded feature. It has been 6 months since the new code has been exercised by default, so it's time to remove the old code path. This issue tracks source code cleanup so we don't forget.

### PR DESCRIPTION
**Pull request by suniltheta**
Commit Message: Replace envoy_reloadable_features_successful_active_health_check_uneject_host with config 
Additional Description: Added a config flag into Outlier Detection to override this behavior to false if required. By default the flag will be considered `true`, to keep the existing behavior where the reloadable feature was default `true`.
Risk Level: Low
Testing: New unit test for non default behavior
Docs Changes: Updated in `docs/root/intro/arch_overview/upstream/outlier.rst`
Release Notes: Included in `changelogs/current.yaml`
Platform Specific Features: NA
Fixes: https://github.com/envoyproxy/envoy/issues/29099
Deprecated Runtime guard: Removed `envoy_reloadable_features_successful_active_health_check_uneject_host`

### CODE REVIEW

**Comment by phlax** [comment]
@suniltheta looks like this needs a `main` merge

/docs

File: docs/root/intro/arch_overview/upstream/outlier.rst [hunk 1]
```diff
@@ -203,8 +203,11 @@ For gRPC requests, the outlier detection will use the HTTP status mapped from th
 
 .. note::
 
-  If :ref: `active health <arch_overview_health_checking>` is also configured, a successful active health check unejects the host and
-  clears all outlier detection counters.
+  If :ref:`active health <arch_overview_health_checking>` is also configured, a successful active health check unejects the host and
+  clears all outlier detection counters. If your active health check is not validating data plane traffic then in situation where
+  active health check passes but the traffic is failing the endpoint will be unejected prematurely. To disable this option then set
+  :ref:`outlier_detection.successful_active_health_check_uneject_host<envoy_v3_api_field_config.cluster.v3.OutlierDetection.successful_active_health_check_uneject_host>`
+  configuration flag as false.
```
Comment (phlax) [inline docs/root/intro/arch_overview/upstream/outlier.rst:210]:
```suggestion
  If :ref:`active health checking <arch_overview_health_checking>` is also configured, a successful active health check unejects the host and
  clears all outlier detection counters. If your active health check is not validating data plane traffic then in situations where
  active health checking passes but the traffic is failing, the endpoint will be unejected prematurely. To disable this option then set
  :ref:`outlier_detection.successful_active_health_check_uneject_host<envoy_v3_api_field_config.cluster.v3.OutlierDetection.successful_active_health_check_uneject_host>`
  configuration flag to ``false``.
```

File: changelogs/current.yaml [hunk 2]
```diff
@@ -44,6 +44,13 @@ bug_fixes:
 - area: tls
   change: |
     fixed a bug where handshake may fail when both private key provider and cert validation are set.
+- area: healthcheck
+  change: |
+    The default behavior of unejecting outlier detection ejected host on successful active health check can
```
Comment (phlax) [inline changelogs/current.yaml:49]:
```suggestion
    The default behavior of unejecting outlier-detection-ejected host on successful active health checking can
```

File: api/envoy/config/cluster/v3/outlier_detection.proto [hunk 3]
```diff
@@ -161,4 +161,10 @@ message OutlierDetection {
   // See :ref:`max_ejection_time_jitter<envoy_v3_api_field_config.cluster.v3.OutlierDetection.base_ejection_time>`
   // Defaults to 0s.
   google.protobuf.Duration max_ejection_time_jitter = 22;
+
+  // If active health check is enabled and a host is ejected by outlier detection, a successful active health check
+  // unejects the host by default and consider it as healthy. Unejection also clears all the outlier detection counters.
```
Comment (phlax) [inline api/envoy/config/cluster/v3/outlier_detection.proto:166]:
```suggestion
  // unejects the host by default and considers it as healthy. Unejection also clears all the outlier detection counters.
```

File: api/envoy/config/cluster/v3/outlier_detection.proto [hunk 4]
```diff
@@ -161,4 +161,10 @@ message OutlierDetection {
...
+  // To change this default behavior set this config to false where active health check will not uneject the host.
```
Comment (phlax) [inline api/envoy/config/cluster/v3/outlier_detection.proto:167]:
```suggestion
  // To change this default behavior set this config to ``false`` where active health checking will not uneject the host.
```

File: api/envoy/config/cluster/v3/outlier_detection.proto [hunk 5]
```diff
@@ -161,4 +161,10 @@ message OutlierDetection {
   // See :ref:`max_ejection_time_jitter<envoy_v3_api_field_config.cluster.v3.OutlierDetection.base_ejection_time>`
   // Defaults to 0s.
   google.protobuf.Duration max_ejection_time_jitter = 22;
+
+  // If active health check is enabled and a host is ejected by outlier detection, a successful active health check
```
Comment (phlax) [inline api/envoy/config/cluster/v3/outlier_detection.proto:165]:
```suggestion
  // If active health checking is enabled and a host is ejected by outlier detection, a successful active health check
```

**Review by phlax — COMMENTED** [review]
thanks @suniltheta - ive left a few grammar/markup nits re docs

**Review by mattklein123 — APPROVED** [review]
LGTM, thank you!

**Pull request merged**